Materials2050_Adapter/Convert/ToBHoM.cs

```diff
@@ -0,0 +1,271 @@
+        {
+            EnvironmentalProductDeclaration epd = new EnvironmentalProductDeclaration();
+            List<EnvironmentalProductDeclaration> epdList = new List<EnvironmentalProductDeclaration>();
+
+            // CustomObjects will vary based on which API is used for the query
+            // Only OpenAPI is implemented right now
+            Enum apiT = config.APIName;
+            if (config != null)
+            {
+                switch (apiT)
+                {
+                    case APIName.Undefined:
+                        {
+                            BH.Engine.Base.Compute.RecordError($"No objects were pulled. {apiT} API was used for the query.");
+                            break;
+                        }
+                    case APIName.OpenAPI:
+                        {
+                            epd = fromOpenAPI(obj);
+                            break;
+                        }
+                    case APIName.GenericMaterialsAPI:
+                        {
+                            BH.Engine.Base.Compute.RecordError($"The {apiT} API has not yet been implemented, no objects were created from the API query. Please use the OpenAPI.");
+                            break;
+                        }
+                    case APIName.ProductAPI:
+                        {
+                            BH.Engine.Base.Compute.RecordError($"The {apiT} API has not yet been implemented, no objects were created from the API query. Please use the OpenAPI.");
+                            break;
+                        }
+                    case APIName.GlobalWarmingAPI:
+                        {
+                            BH.Engine.Base.Compute.RecordError($"The {apiT} API has not yet been implemented, no objects were created from the API query. Please use the OpenAPI.");
+                            break;
+                        }
+                    default:
+                        break;
+                }
+                return epd;
+            }
+            else
+            {
+                BH.Engine.Base.Compute.RecordError("Error.");
+                return null;
+            }
+        }
+
+        /***************************************************/
+
+        public static EnvironmentalProductDeclaration fromOpenAPI(CustomObject co)
+        {
+            EnvironmentalProductDeclaration epd = new EnvironmentalProductDeclaration();
+            // OpenAPI call object schema
+
+            // basic results parsing data
+            ResultsNavigation resultsNavigation = new ResultsNavigation();
+
+            resultsNavigation.TotalProducts = (int)co.PropertyValue("TotalProducts");
+            resultsNavigation.ProductsOnPage = (int)co.PropertyValue("countProductsOnPage");
+            resultsNavigation.CurrentPage = (int)co.PropertyValue("current_page");
+            resultsNavigation.ProductRange = co.PropertyValue("product_range").ToString();
```

Comment:
This will throw errors if any of the objects cannot be cast to `int` or is `null` and cannot call `ToString` on it.

Does not _need_ fixing in this PR, can be done later - just raising as a flag - but this is prototype so not crucial but if you start getting errors this might be a place to put some protections in.

Reply:
fixed in previous commit. Although it's extremely shorthand ternarys so not super easy to read for everyone. Happy to pick up in a subsequent PR.